go/pkg/pwagent/nginx.go

```diff
@@ -250,8 +254,20 @@ func Nginx(ctx context.Context, opts AgentOpts, cli *client.Client, logger *zap.
 
 					// add hash per users to proxy configuration
 					if _, found := opts.AllowedUsers[instance.Names[0][1:]]; found {
-						for _, hash := range opts.AllowedUsers[instance.Names[0][1:]] {
-							upstream.Hashes = append(upstream.Hashes, strconv.Itoa(int(hash)))
+						for _, userID := range opts.AllowedUsers[instance.Names[0][1:]] {
+							stringToHash := fmt.Sprintf("%s%d%s", instance.ID, userID, opts.Salt)
+							hashBytes := make([]byte, 8)
+							hasher := sha3.NewCShake256([]byte("NAME"), []byte("Partition1"))
+							_, err = hasher.Write([]byte(stringToHash))
+							if err != nil {
+								return errcode.ErrNotImplemented.Wrap(err)
```

Comment:
```suggestion
								return errcode.TODO.Wrap(err)
```

When you don't have an explicit error, use `errcode.TODO` (it's not a problem to have some ones merged into master, especially for tools that are only exposed to admins or challenge developers)